### PR TITLE
Update version strings for release

### PR DESCRIPTION
**Pull request by gonuke**
Updated version strings for release in two canonical places.

* [x] Still need to update pyne submodule.

### CODE REVIEW

**Comment by shimwell** [comment]
regarding the outstanding task, is that similar to the submodules that Bam was working on in PR #742

**Comment by gonuke** [comment]
> regarding the outstanding task, is that similar to the submodules that Bam was working on in PR #742

I don't think so???? We simply need to advance the hash that our submodule refers to now that we have a hash that points to a release (candidate)

**Comment by gonuke** [comment]
> > regarding the outstanding task, is that similar to the submodules that Bam was working on in PR #742
> 
> I don't think so???? We simply need to advance the hash that our submodule refers to now that we have a hash that points to a release (candidate)

However, I need to wait for the PyNE RC PR to be merged

**Comment by shimwell** [comment]
> > > regarding the outstanding task, is that similar to the submodules that Bam was working on in PR #742
> > 
> > 
> > I don't think so???? We simply need to advance the hash that our submodule refers to now that we have a hash that points to a release (candidate)
> 
> However, I need to wait for the PyNE RC PR to be merged

Just merged PyNE RC 0.7.6

**Comment by gonuke** [comment]
I just updated PyNE but noticed that we don't run a full set of CI on that change.... stay tuned...

**Comment by gonuke** [comment]
Good thing we're waiting - the new PyNE version string stuff broke DAGMC builds!

File: src/uwuw/tests/uwuw_unit_tests.cpp
Comment (pshriwise) [inline src/uwuw/tests/uwuw_unit_tests.cpp:151]: Thank you for taking care of this! Saw this was why the tests were failing and meant to send a PR your way but you beat me to it.

**Comment by gonuke** [comment]
Since this is a PR to a non-`develop` branch, it doesn't run tests, but successful tests can be found here: https://github.com/gonuke/DAGMC/actions

File: release.rst
Comment (pshriwise) [inline release.rst]:
```suggestion
1. The DAGMC repository has a `stable` Docker image tag that points to the same SHA as the newest release.
```

File: release.rst
Comment (pshriwise) [inline release.rst]:
```suggestion
2. The PyNE CI docker images rely on the DAGMC `stable` Docker image tag (this was incorporated in PyNE PR #1415)
```

File: release.rst
Comment (pshriwise) [inline release.rst]:
```suggestion
    b. Create a pull request (PR) *into the PyNE RC branch* in order to
```

File: release.rst
Comment (pshriwise) [inline release.rst]:
```suggestion
    d. Potentially perform other updates to the RC branch via PRs
```

File: release.rst
Comment (pshriwise) [inline release.rst]:
```suggestion
           branch. The PyNE RC branch is still not merged at this point.
```

File: release.rst
Comment (pshriwise) [inline release.rst]:
```suggestion
        ii. Update the DAGMC version number in `CMakeLists.txt`
```

File: release.rst
Comment (pshriwise) [inline release.rst]:
```suggestion
        iii. Update DAGMC's `CHANGELOG.rst`
```

File: release.rst
Comment (pshriwise) [inline release.rst]:
```suggestion
   will automatically build & test PyNE with the updated DAGMC RC
```

**Review by pshriwise — CHANGES_REQUESTED** [review]
A few suggestions for the release documentation. Thank you for putting that together, @gonuke.

File: release.rst
Comment (gonuke) [reply release.rst]: actually - this is a tag in the source code repo - but perhaps I should still clarify

File: release.rst
Comment (gonuke) [reply release.rst]: ibid

File: release.rst
Comment (gonuke) [reply release.rst]: because it is the `stable` tag - I'll clarify

File: release.rst
Comment (gonuke) [reply release.rst]: I'm actually not sure how to clarify this without being very wordy.  Maybe we should use something other than `stable` as the tag since this so often is used to specify a docker image

File: release.rst
Comment (shimwell) [reply release.rst]: We could add a [link](https://github.com/svalinn/DAGMC/pkgs/container/dagmc-ci-ubuntu-20.04) to the docker image is that helps

**Review by shimwell — APPROVED** [review]
I think all the comments have been addressed and this looks ready to me (with my terrible c++ skills). I've read through for typos and easy to spot errors. It is really nice to see the release details documented. This PyNe DAGMC interdependency makes the process quite tricky.

File: release.rst
Comment (pshriwise) [reply release.rst]: Ah, my mistake. `stable` makes sense to me as long as we clarify.

**Review by pshriwise — APPROVED** [review]
Looks good @gonuke!

**Comment by pshriwise** [comment]
Will merge at EOB today if there are no further comments.

**Pull request merged**
